assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -248,16 +257,63 @@ const baseActions = {
 			settingsEditComponent,
 			settingsViewComponent,
 			setupComponent,
+			checkRequirements,
 		};
 
-		return {
+		yield {
 			payload: {
 				settings,
 				slug,
 			},
 			type: REGISTER_MODULE,
 		};
+
+		const registry = yield Data.commonActions.getRegistry();
+
+		// As we can specify a custom checkRequirements function here, we're invalidating the resolvers for activation checks.
+		yield registry.dispatch( STORE_NAME ).invalidateResolution( 'canActivateModule', [ slug ] );
+		yield registry.dispatch( STORE_NAME ).invalidateResolution( 'getCheckRequirementsError', [ slug ] );
+	},
+
+	/**
+	 * Receives the check requirements error map for specified modules modules.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} slug  Module slug.
+	 * @param {Object} error WordPress Error object containing code, message and data properties.
+	 * @return {Object} Action for RECEIVE_CHECK_REQUIREMENTS_ERROR.
+	 */
+	receiveCheckRequirementsError( slug, error ) {
+		invariant( slug, 'slug is required' );
+		invariant( error, 'error is required' );
+		invariant( isPlainObject( error ), 'error must be an object' );
+		return {
+			payload: { slug, error },
+			type: RECEIVE_CHECK_REQUIREMENTS_ERROR,
+		};
+	},
+
+	/**
+	 * Receives the check requirements success for a module.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {string} slug Success for a module slug.
+	 * @return {Object} Action for RECEIVE_CHECK_REQUIREMENTS_SUCCESS.
+	 */
+	receiveCheckRequirementsSuccess( slug ) {
+		invariant( slug, 'module slug is required' );
```

Comment:
Let's just say `slug is required` according to the variable name, like above too.

Reply:
Addressed.